example/lib/ios_popover/_ios_popover_menu.dart

```diff
@@ -0,0 +1,222 @@
+  final EdgeInsets padding;
+
+  /// Color of the decoration.
+  final Color backgroundColor;
+
+  /// Center point of the arrow.
+  ///
+  /// Defaults to the center of the axis.
+  final double? arrowFocalPoint;
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return DecoratedBox(
+      decoration: ShapeDecoration(
+        color: backgroundColor,
+        shape: _IosMenuShapeBorder(
+          arrowDirection: arrowDirection,
+          arrowFocalPoint: arrowFocalPoint,
+          arrowLength: arrowLength,
+          arrowWidth: arrowWidth,
+          padding: padding,
+          radius: radius,
+        ),
+      ),
+      child: Padding(
+        padding: _getArrowPadding(),
+        child: Padding(
+          padding: padding,
+          child: child,
+        ),
+      ),
+    );
+  }
+
+  /// Returns the padding needed to leave space for the arrow.
+  EdgeInsets _getArrowPadding() {
+    return EdgeInsets.fromLTRB(
+      arrowDirection == ArrowDirection.left ? arrowLength : 0,
+      arrowDirection == ArrowDirection.up ? arrowLength : 0,
+      arrowDirection == ArrowDirection.right ? arrowLength : 0,
+      arrowDirection == ArrowDirection.down ? arrowLength : 0,
+    );
+  }
+}
+
+class _IosMenuShapeBorder extends ShapeBorder {
+  const _IosMenuShapeBorder({
+    required this.radius,
+    required this.arrowWidth,
+    required this.arrowLength,
+    required this.arrowDirection,
+    required this.padding,
+    this.arrowFocalPoint,
+  });
+
+  final Radius radius;
+  final double arrowWidth;
+  final double arrowLength;
+  final ArrowDirection arrowDirection;
+  final EdgeInsets padding;
+  final double? arrowFocalPoint;
+
+  @override
+  EdgeInsetsGeometry get dimensions => EdgeInsets.zero;
+
+  @override
+  Path getInnerPath(Rect rect, {TextDirection? textDirection}) => Path();
+
+  @override
+  Path getOuterPath(Rect rect, {TextDirection? textDirection}) {
+    return _IosDecorationPathBuilder(
+      size: rect.size,
+      arrowDirection: arrowDirection,
+      arrowLength: arrowLength,
+      arrowWidth: arrowWidth,
+      radius: radius,
+      arrowFocalPoint: arrowFocalPoint,
+    ).build().shift(Offset(rect.left, rect.top));
+  }
+
+  @override
+  void paint(Canvas canvas, Rect rect, {TextDirection? textDirection}) {}
+
+  @override
+  ShapeBorder scale(double t) => this;
+}
+
+/// Builds a path used to paint an iOS popup menu decoration.
+class _IosDecorationPathBuilder {
+  _IosDecorationPathBuilder({
+    required this.size,
+    required this.arrowLength,
+    required this.arrowWidth,
+    required this.arrowDirection,
+    required this.radius,
+    this.arrowFocalPoint,
+  });
+
+  final Size size;
+  final double arrowLength;
+  final double arrowWidth;
+  final ArrowDirection arrowDirection;
+  final Radius radius;
+  final double? arrowFocalPoint;
+
+  Path build() {
+    // Width of each side of the arrow.
+    final arrowStep = arrowWidth / 2;
```

Comment:
I'm not sure what either of these phrases mean: "width of each side of the arrow", "arrowStep". Neither of those sound like aspects of a triangle.